NewForce-Cohort-5/tabloidfullstack-the-pollening
# AE #9 View All Categories

# Description

This is for Ticket#9(https://trello.com/c/HnaVNqZv/9-9-view-all-categories), which lets a user with the admin privilege see all the available categories.  
# Type of change
New feature (non-breaking change which adds functionality)


List of changes made

(Client-Side)
category route path was added to ApplicationViews.js, along with wrapping that route with a CategoryProvider
category  route was added to Header.js in the navbar as a navlink to view all categories
Components were built out for Category.js and CategoryList.js
Category.css added
Provider was built for CategoryProvider.js

(Server-Side)
Category model for the C# category class
Category repository and ICategory repository to have a private class CategoryRepositry and its method to get all categories from the sql database to the c# class 
Updated the ConfigureServices in startup.js  
Category controller so that client side can talk to the server side

# Testing Instructions

**Testing objectives**
1.  To see whether the app will direct to the Category page when a user  select the `Category Management` link in the menu
2. To see whether the app will alphabetically display all the available categories at the category page


**Testing instructions**
(This is the similar testing instructions from the previous merged pull request.  Updated parts are **bold**.  If you have already followed these instructions below, please do **No.2**, **parts of No.13**,      through No.16) 

1. In Terminal, cloning down this repository to your local repository.
2. If you currently do not have access to this branch, run a <git fetch --all>, then navigate to this branch **`ae-viewAllCategories`** by typing git checkout **`ae-viewAllCategories`** in Terminal. And type Start Tabloid.sln in Terminal to run this program through Visual Studio 2022.
3. If you have not created or seeded the basic database, you will need to do so. First, open a new query by right clicking on Tabloid under the database file in the SQL Server Object Explorer on the left.
4. You will need to create the table first and then add the seed data. Choose files in the following order:
5. [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. To run the query, please click the green outlined arrow under the query tab.
8. Please run the following sql commands to check the database tables have been created  and to check the data is in the   the UserProfile table

  ```
SELECT  * FROM
INFORMATION_SCHEMA.TABLES;
GO
Select * from UserProfile;
```

9. To run the back end server, please change the solid green debugger database from IISExpress to Tabloid and click the solid green debugger play button.
10. The swagger UI should run in a browser.
11. And then on the client side, please update the value of the variable called apiUrl to "https://localhost:5001/" In the UserProfile.js
12. Execute the npm start command on the terminal at the level where you can see package.json
13. A browser should start to run the program with “localhost” as a part of the URL. The opening page should be the Log-in page. Please log in using this email address that should be in your database to log in: [foo@bar.com](mailto:foo@bar.com)
14. Please select Category in the header and to see if the app direct you to the category page(Testing objective1)
15. Please check to see if categories are displayed alphabetically when you select category on the header. (Testing objective 2)
16. If there are no issues, please go back to Terminal and run the following command ‘git checkout main’ to get back to the local main branch



# Checklist:
